model/application.go

```diff
@@ -40,6 +48,52 @@ type Application struct {
 	// read only: true
 	// required: true
 	// example: image/image.jpeg
-	Image    string            `gorm:"type:text" json:"image"`
-	Messages []MessageExternal `json:"-"`
+	Image    string               `gorm:"type:text" json:"image"`
+	Messages []ApplicationMessage `json:"-"`
+}
+
+// ApplicationMessage Model
```

Comment:
Not really into the code yet, but why do we need an ApplicationMessage besides the normal Message?

Reply:
I saw that the `MessageExternal` struct was used for two different things, namely
- encoding of messages _sent_ to a client,
- encoding of messages _received_ from an application.

Since these things are fundamentally different, I felt the need to create a dedicated struct for the latter. I can remove it if you whish it to be excluded from this pull request. However, since the `CreateMessage` is a bit cleaner by using internal messages, I thought it made sense to include it here.

Also, note the commit message for 20a4510ee7e95bc0bd196b7979fb47da0ac15c9c:
> We introduce the struct ApplicationMessage that is used for receiving new messages from applications. Since applications cannot specify a message ID or date, this ensures that the message sent from applications does not falsely include such members.

In short, the `ApplicationMessage` does not have members that are not accepted in messages sent by applications. Previously, we accepted these fields, but omitted them while processing the message.

Reply:
Alright, Maybe we name it InputMessage or something similar, that it's clear what it does?

Reply:
I renamed it. Can we close this conversation?